spec/components/guardian_spec.rb

```diff
@@ -2035,6 +2035,34 @@
 
   end
 
+  context "can_delete_post?" do
+    fab!(:post) { Fabricate(:post, post_number: 2) }
+
+    it "returns false for regular users, except poster" do
+      expect(Guardian.new(post.user).can_delete?(post)).to eq(true)
```

Comment:
There are some duplicated assertions with https://github.com/discourse/discourse/blob/a6363170e9c91384292bd4fa5f497803a4eae906/spec/components/guardian_spec.rb#L2123-L2189. We should probably group these tests together.